static/js/components/VegaPlot.jsx

```diff
@@ -166,20 +166,36 @@ const spec = (url) => ({
 });
 
 
-const VegaPlot = ({ dataUrl }) => (
-  <div
-    ref={
-      (node) => {
-        embed(node, spec(dataUrl), {
-          actions: false
-        });
-      }
-    }
-  />
-);
+class VegaPlot extends React.Component {
+  // This is implemented as a class so we can define
+  // shouldComponentUpdate
```

Comment:
Can we use hooks instead?

https://stackoverflow.com/a/56297625/214686

Reply:
Done.